docs/serving/using-knative-services.md

```diff
@@ -0,0 +1,233 @@
+---
+title: "User-created Knative Services"
+weight: 01
+linkTitle: "User-created Knative Services"
+type: "docs"
+---
+
+Knative services are Kubernetes Services that a user creates to deploy an application. Each Knative Service is defined by a Route and a Configuration, which have the same name as the Service, contained in a YAML file.
+
+## Before you begin
+
+To use Knative Services, you will need:
+* A Kubernetes cluster with [Knative installed](https://knative.dev/docs/install/index.html).
+
+## Creating a Knative Service
+To create an application, you need to create a YAML file that defines a Knative Service.
+This YAML file specifies metadata about the application, points to the hosted image of the app for deployment, and allows the deployment to be configured.
+
+This guide uses the [Hello World sample app in Go](https://knative.dev/docs/serving/samples/hello-world/helloworld-go) to demonstrate the structure of a Service YAML file and the basic workflow for deploying an app, but these steps can be adapted for your own application if you have an image of it available on Docker Hub, Google Container Registry, or another container image registry.
+
+The Hello World sample app does the following:
+1. reads in an env variable, TARGET, from the configuration .yaml file
+2. prints “Hello World: \${TARGET}!“. If TARGET isn’t defined, it will print “NOT SPECIFIED”.
+
+### Procedure
+
+1. Create a new file named `service.yaml` containing the following information.
```

Comment:
To make it easier to follow you could make this a [HereDoc](https://en.wikipedia.org/wiki/Here_document) that outputs to service.yaml so all I need to do is copy the command and press enter.

Reply:
First time I've heard of this :slightly_smiling_face: @RichieEscarez do you know if this is something we're using and if so, do we have any contributor docs for how to do this? Thanks.

Reply:
@dgerd is this an example of what your are proposing (HereDoc), here in one of our other topics: https://knative.dev/docs/eventing/samples/kafka/channel/#creating-a-kafkachannel-channel-crd ?

ie. Where you wrap the [YAML] with with the following so that users can edit, apply, and deploy in a single command?

```
cat <<-EOF | kubectl apply -f -
---
[ALL_THE_YAML_GETS_COPIED_HERE?]
EOF
```

Question: This syntax does not create a copy of that file locally for the user, right? Is there a way to also create a local file?

@abrennan89 today, we have the [`readfile` shortcode](https://github.com/knative/website/blob/master/layouts/shortcodes/readfile.md) that will copy in an actual code file (at markdown build time) so that you dont have to manually copy and hardcode them into the markdown. Example here https://knative.dev/smoketest/

`{{< readfile file="./docs/serving/samples/hello-world/helloworld-go/service.yaml" code="true" lang="yaml" >}}`

Per @dgred's feedback, maybe I could open a PR to add an option to `readfile` (another flag: `heredoc="true"`) so that it automatically adds the "HereDoc" syntax around that code?

Reply:
> This syntax does not create a copy of that file locally for the user, right? Is there a way to also create a local file?

This syntax does not. `tee` is a common command that can be inserted between to make a local copy. Alternatively the file creation through a heredoc and the application through kubectl could be broken into two steps if storing the file locally is desired.